chain/types/ethtypes/errors.go

```diff
@@ -0,0 +1,44 @@
+package ethtypes
+
+import (
+	"fmt"
+
+	"golang.org/x/xerrors"
+)
+
+var ErrExecutionReverted = xerrors.New("execution reverted")
+
+// ExecutionRevertedError is an error that occurs when a transaction reverts.
+type ExecutionRevertedError struct {
+	Message string
+	Code    int
+	Meta    []byte
+	Data    string
+}
+
+// Error implements the error interface.
+func (e *ExecutionRevertedError) Error() string {
+	if e.Message == "" {
+		return fmt.Sprintf("json-rpc error %d", e.Code)
+	}
+	return e.Message
+}
+
+// ErrorData returns the error data.
+func (e *ExecutionRevertedError) ErrorData() interface{} {
+	return e.Data
+}
+
+// ErrorCode returns the JSON error code for a revert.
+func (e *ExecutionRevertedError) ErrorCode() int {
+	return 3
```

Comment:
Why is this hard coded to 3 ?

Reply:
Don't we have a `Code` field on `ExecutionRevertedError` ? Shouldn't we use that ?

Reply:
Yeah this (3) should be default error code for `EthCall` and `EstimateGas`. Will update it.

Reply:
Yeah that make sense, So let me confirm once what kind of error codes FVM returns and I will update it accordingly.